File: python/easy/(115) mixed-content.py
import sys

def main(filepath):
    with open(filepath, 'r') as f:
        for line in f.readlines():
            if line:
                line = line.strip()
                line = line.split(',')
                
                word_list = []
                digit_list = []

                for elm in line:
                    if elm.isdigit():
                        digit_list.append(elm)
                    else:
                        word_list.append(elm)

                if word_list and digit_list:
                    print ('%s|%s') % (','.join(word_list), ','.join(digit_list))
                elif word_list and not digit_list:
                    print ','.join(word_list)
                elif not word_list and digit_list:
                    print ','.join(digit_list)

              
if __name__ == '__main__':
    main(sys.argv[1])

File: java/easy/(115) mixed-content.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            
            String[] lineArray = line.split(",", -1);
            
            List<String> wordList = new ArrayList<>();
            List<String> digitList = new ArrayList<>();

            for (String elm : lineArray){
                if (Main.isNumeric(elm)){
                    digitList.add(elm);
                }
                else{
                    wordList.add(elm);
                }
            }

            String result = "";

            if (wordList.size() > 0){
                for (String word : wordList){
                    result += word + ",";
                }
                result = result.substring(0, result.length()-1);
            }

            if ((wordList.size()) > 0 && (digitList.size() > 0)){
                result += "|";
            }

            if (digitList.size() > 0){
                for (String digit : digitList){
                    result += digit + ",";
                }
                result = result.substring(0, result.length()-1);
            }

            System.out.println(result);

        }
    }

    public static boolean isNumeric(String inputData) {
        return inputData.matches("[-+]?\\d+(\\.\\d+)?");
    }
}